telemetry/packages/server/src/modules/openmct/faults/Fault.service.ts

```diff
@@ -56,6 +56,146 @@ export class FaultService {
+
+    const point = new Point('fault')
+      .timestamp(now)
+      .tag('faultId', updatedFault.fault.id)
+      .tag('podId', fault.podId)
+      .tag('measurementKey', fault.measurementKey)
+      .stringField('fault', JSON.stringify(updatedFault));
+
+    try {
+      this.influxService.faultsWrite.writePoint(point);
+
+      this.logger.debug(
+        `Acknowledged fault with id ${updatedFault.fault.id}`,
+        FaultService.name,
+      );
+    } catch (e: unknown) {
+      this.logger.error(
+        `Failed to acknowledge fault with id ${updatedFault.fault.id}`,
+        e,
+        FaultService.name,
+      );
+    }
+
+    this.realtimeService.sendFault(updatedFault);
+  }
+
+  /**
+   * Shelves or unshelves a fault in the database and sends it to the client.
+   * @param faultId The id of the fault to shelve
+   * @param shelved Whether to shelve or unshelve the fault
+   * @param shelveDuration The duration to shelve the fault for (in milliseconds)
+   * @param comment The comment sent with the shelve
+   */
+  public async shelveFault(
+    faultId: string,
+    shelved: boolean,
+    shelveDuration: number,
+    comment?: string,
+  ) {
+    this.logger.debug(
+      `Shelving fault with id ${faultId} for ${shelveDuration} seconds. ${comment ? `Comment: ${comment}` : ''}`,
+      FaultService.name,
+    );
+
+    // Get the fault from the database
+    const possibleFault = await this.historicalService.getHistoricalFaults({
+      faultId,
+    });
+
+    if (!possibleFault || possibleFault.length === 0) {
+      this.logger.error(
+        `Fault with id ${faultId} not found`,
+        FaultService.name,
+      );
+      return;
+    }
+
+    const fault = possibleFault[0];
+    const updatedFault = fault.openMctFault;
+    updatedFault.fault.shelved = shelved;
+
+    const now = new Date(Date.now());
+
+    const point = new Point('fault')
+      .timestamp(now)
+      .tag('faultId', updatedFault.fault.id)
+      .tag('podId', fault.podId)
+      .tag('measurementKey', fault.measurementKey)
+      .stringField('fault', JSON.stringify(updatedFault));
+
+    // If the fault is being shelved, set a timer to unshelve it
+    if (shelved) {
+      this.logger.debug(
+        `Setting timer to unshelve fault with id ${faultId}`,
+        FaultService.name,
+      );
+      // This isn't ideal, since this doesn't persist if the server restarts.
+      // Also, if the fault is manually unshelved before the timer runs out, the timer should be cancelled and it isn't.
+      // (So the fault will be unshelved twice, once manually and once automatically. Plus, if the fault is unshelved and then shelved again, the timer won't be reset.)
+      setTimeout(() => {
+        this.logger.debug(
+          `Automatically unshelving fault with id ${faultId} because the shelve duration has passed`,
+          FaultService.name,
+        );
+        void this.shelveFault(faultId, false, 0, '');
+      }, shelveDuration);
+    }
```

Comment:
Timers could be stored in a global timers object then cleared after manual or automatic unshelving.
`timers: Record<string, Node.JSTimeout`
...
`this.timers[faultId] && clearTimeout( this.timers[faultId] )`
`if (shelved) {`
`    this.timers[faultId] = setTimeout(() => ... )`
`}`

Reply:
Good idea, this would solve the manual shelving issue. Issue would still persist on server restarts though. I suspect the best option would be to store the unshelve time in the database - will open an issue for this and we can sort it separately :)